.grit/patterns/re_match_walrus.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Walrus operator for match in if
+---
+
+Use the walrus operator for snippets with the pattern:
+
+```python
+# Convert this:
+math = re.match(regex):
+if match:
+   ...
+
+# To this:
+if math := re.match(regex):
+   ...
+```
+
+Limitations:
+* If the match function is imported with an alias (e.g. `from re import match as m`), it will not be transformed.
+* When `re.match` is used, we do not check that `re` comes from `import re`.
+
+
+```grit
+engine marzano(0.1)
+language python
+
+pattern re_match_function() {
+    or {
+        `search`,
+        `match`,
+        `fullmatch`,
+    }
+}
+
+pattern imported_match_function() {
+    $func where {
+        $func <: re_match_function(),
+        $func <: imported_from(source = `re`),
+    },
+}
+
+// TODO: we should also check if `re` is imported
```

Comment:
Not sure I follow the difference here. Either way please resolve this todo and avoid commented out code.

Reply:
I've removed the comment. I was thinking of double-checking that `re` is an imported module and not something else, but I guess it's nitpicking.